subdir/subdir.go

```diff
@@ -33,7 +33,7 @@ func (s *subdirFs) Create(filename string) (billy.File, error) {
 		return nil, err
 	}
 
-	return newFile(f, filename), nil
+	return newFile(s, f, filename), nil
```

Comment:
Maybe this logic inside the `newFile` method (`filename = fs.Join(fs.Base(), filename)`) should be outside. Then the `newFile` method can only receive a `billy.File` and a `path`.